### PR TITLE
Add DaemonSet deploy option in Helm Chart

### LINKED ISSUE #600: Provide daemonset deployment mode in Capsule Helm Chart
# Describe the feature
Deploying Capsule to the cluster is done through the usual means, but for minimal latency in the requests, it might be a good idea to let a `daemonset` deploy the Capsule pods to the same master nodes as the API servers run on.

Since multiple replicas of Capsule is supported, it would be useful to have `daemonset` deploy mode in the Helm Chart too.

# What would the new user story look like?
The cluster admin uses daemonset deployment mode for Capsule Operator on the same nodes where APIs server is running.

# Expected behavior
Capsule Operator is deployed on the same nodes where APIs server is running

### PR DESCRIPTION
**Pull request by bsctl**
close #600

### CODE REVIEW

File: charts/capsule/templates/pre-delete-job.yaml [hunk 1]
```diff
@@ -1,4 +1,7 @@
-{{- $cmd := printf "kubectl scale deployment -n $NAMESPACE %s --replicas 0 &&" (include "capsule.deploymentName" .) -}}
+{{- $cmd := ""}}
+{{- if eq .Values.manager.kind "Deployment" }}
+{{- $cmd = printf "kubectl scale deployment -n $NAMESPACE %s --replicas 0 &&" (include "capsule.controllerName" .) -}}
```
Comment (bsctl) [inline charts/capsule/templates/pre-delete-job.yaml:3]: @prometherion sorry, not understand the reason to scale down before delete

Comment (prometherion) [reply charts/capsule/templates/pre-delete-job.yaml:3]: Not sure, I think we can drop it.

**Review by prometherion — CHANGES_REQUESTED** [review]
Let's give a try by removing the scale down upon deletion.

File: charts/capsule/templates/pre-delete-job.yaml (hunk 1 above)
Comment (bsctl) [reply charts/capsule/templates/pre-delete-job.yaml]: It seems no effects, dropping it.

**Review by prometherion — APPROVED** [review]

**Pull request merged**